honeybee_energy/ventcool/afn.py

```diff
@@ -0,0 +1,391 @@
+    reference crack data in int_cracks, by the wall area.
+
+    This function does not check adjacency information when computing leakage parameters
+    for adjacent faces or subfaces. It assumes adjacent faces share the same type and
+    area properties so that the computed leakage parameters are equivalent. Note that
+    EnergyPlus only requires one of the adjacent interzone surfaces to be
+    assigned a leakage component. If both adjacent surfaces have a leakage component,
+    the air flow through the surface will be counted twice.
+
+    Args:
+        interior_face_groups: A tuple with four groups of interior faces types:
+
+            * int_walls: List of interior Wall type Face objects.
+            * int_floorceilings: List of interior RoofCeiling and Floor type Face
+                objects.
+            * int_apertures: List of interior Aperture subface Face objects.
+            * int_doors: List of interior Door subface Face objects.
+
+        int_cracks: A dictionary of air mass flow coefficient and exponent data
+            corresponding to the face types in the interior_face_groups. Face
+            data flow coefficients should be normalized by surface area, and closed
+            opening flow coefficients should be normalized by edge lengths, for example:
+
+            {
+                # kg/s/m2 @ 1 Pa
+                'wall_flow_cof': 0.003,
+                'wall_flow_exp': 0.75,
+                'floorceiling_flow_cof': 0.0009,
+                'floorceiling_flow_exp': 0.7,
+                # kg/s/m @ 1 Pa
+                'window_flow_cof': 0.0014,
+                'window_flow_exp': 0.65,
+                'door_flow_cof': 0.02,
+                'door_flow_exp': 0.6
+            }
+    """
+
+    # simplify parameters
+    int_walls, int_floorceilings, int_apertures, int_doors = interior_face_groups
+
+    # add interior crack leakage components
+    for int_wall in int_walls:
+        opening_area = sum([aper.area for aper in int_wall.apertures])
+        opening_area += sum([door.area for door in int_wall.doors])
+        face_area = int_wall.area - opening_area
+        flow_cof = int_cracks['wall_flow_cof'] * face_area
+        flow_exp = int_cracks['wall_flow_exp']
+        int_wall.properties.energy.vent_crack = AFNCrack(flow_cof, flow_exp)
+
+    for int_floorceiling in int_floorceilings:
+        opening_area = sum([aper.area for aper in int_floorceiling.apertures])
+        face_area = int_floorceiling.area - opening_area
+        flow_cof = int_cracks['floorceiling_flow_cof'] * face_area
+        flow_exp = int_cracks['floorceiling_flow_exp']
+        int_floorceiling.properties.energy.vent_crack = AFNCrack(flow_cof, flow_exp)
+
+    # add interior opening leakage components
+    for int_aperture in int_apertures:
+        if int_aperture.properties.energy.vent_opening is None:
+            int_aperture.is_operable = True
+            int_aperture.properties.energy.vent_opening = VentilationOpening()
```

Comment:
Why are you removing the degree of freedom here?  If you need a VentilationOpening just to set infiltration, then set the operable area to 0.